ote/src/clj/ote/services/reports.sql

```diff
@@ -174,21 +174,23 @@ select
 select
   "sub-type",
   count("sub-type") as count
-  from (select distinct top.id as toid,ts."sub-type" from "transport-service" ts, "transport-operator" top where top.id = ts."transport-operator-id") as unusedname
+  from (select distinct top.id as toid,ts."sub-type" from "transport-service" ts, "transport-operator" top where top.id = ts."transport-operator-id" and ts."sub-type" is not null) as unusedname
   group by "sub-type";
 
--- name: create-temp-view-for-monthly-producer-counts-by-sub-type!
-create or replace temporary view
-    monitor_producers_by_type as
-    select to_char(gr.created, 'YYYY-MM') as month,
-    ts.id as tsid,
-    top.id as toid,
-    ts."sub-type"
-    from
-    "transport-service" ts,
-    "transport-operator" top,
-    "group" gr
-    where top.id = ts."transport-operator-id" and gr.id = top."ckan-group-id" and gr.created is not null;
-
--- name: monthly-producer-counts-by-sub-type
-select month, "sub-type", count("sub-type") from monitor_producers_by_type group by "sub-type", month order by month, "sub-type";
+-- subquery name legend is: stq = subtype grouping query, biq = business-id grouping query
+-- 
+-- name: monthly-producer-types-and-counts
```

Comment:
In nap we have used sql river formatting model. You could reformat this file accordingly: https://www.sqlstyle.guide/